compiler/noirc_frontend/src/hir/resolution/errors.rs

```diff
@@ -430,8 +430,8 @@ impl<'a> From<&'a ResolverError> for Diagnostic {
             ResolverError::UseExplicitNumericGeneric { ident } => {
                 let name = &ident.0.contents;
 
-                Diagnostic::simple_warning(
-                    String::from("Noir now supports explicit numeric generics. Support for implicit numeric generics will be removed in the following release."), 
+                Diagnostic::simple_error(
```

Comment:
Probably can delete this error entirely?

Reply:
This is now the only error that informs users that they need to use a numeric generic (or where) and it triggers in different places now.

Reply:
We still need an error but it could be a TypeKindMismatch